runtime/doc/various.txt

```diff
@@ -14,11 +14,13 @@ Various commands					*various*
 
 							*CTRL-L*
 CTRL-L			Clear and redraw the screen.  The redraw may happen
-			later, after processing typeahead.
+			later, after processing typeahead. Terminal buffers
+			fit the embedded terminal into the current window.
 
 							*:redr* *:redraw*
 :redr[aw][!]		Redraw the screen right now.  When ! is included it is
-			cleared first.
+			cleared first and terminal buffers fit the embedded
+			terminal into the current window.
```

Comment:
> if a terminal is shown in multiple windows, which one takes precedence

Only the current buffer/window is taken into account. If there are 10 windows, 9 show terminals, and you're currently in the window without a terminal, `<c-l>` won't resize any terminal.